apm-agent-plugins/apm-log-correlation-plugin/src/main/java/co/elastic/apm/agent/mdc/MdcActivationListener.java

```diff
@@ -158,6 +159,9 @@ public void beforeActivate(TraceContextHolder<?> context) throws Throwable {
                         put.invoke(TRACE_ID, traceContext.getTraceId().toString());
                         put.invoke(TRANSACTION_ID, traceContext.getTransactionId().toString());
                     }
+                    if (context instanceof ErrorCapture) {
```

Comment:
to support errors that don't happen within a transaction:
```java
                    if (context instanceof ErrorCapture) {
                        put.invoke(ERROR_ID, traceContext.getId().toString());
                    } else if (tracer.getActive() == null) {
                        TraceContext traceContext = context.getTraceContext();
                        put.invoke(TRACE_ID, traceContext.getTraceId().toString());
                        put.invoke(TRANSACTION_ID, traceContext.getTransactionId().toString());
                    }
```

Reply:
added